File: individual-projects/2023/amoli-patel-200103102100/Fundamental Practice/loop/20-20_game.js
//Create a loop for 20 20 game. Stop the game if its raining
function main() {
    let n = 0,
        rain // n is a over of match
    while (n < 20) {
        if (rain == true) {
            let total_over = 20
            while (n <= total_over) {
                if (rain == true) {
                    // random over this argument is true
                    console.log('Stop rain is start..')
                    break
                } else {
                    console.log('Play Game...')
                }
                n += 1
            }

            rain = false
            while (n <= total_over) {
                console.log('Continue after rain...')
            }
        }
    }
}

File: individual-projects/2023/ronak-bagdai-200103041003/Fundamental-Practice/Loop/20-20_Game.rs
// fn main() {
//     let n = 20; // 20 overs
//     for i in 1..=n {
//         if rain {
//             break;
//         }
//        // code 1
//        // code 2
//     }

// }








// // Create a loop for 20 20 game. Stop the game if its raining

// let overs = 20;
// let current_score = 0;
// let balls = 0;
// let wickets = 0;
// let is_raining = true;

// fn main(){
//     let result = game(balls,overs, wickets);    
// }


// fn game(balls, overs, wickets){
//     while (balls < overs * 6  && wickets < 10 && !checkIfRaining()):
//             balls++;
//             ball_result = ["dot ball","single","double","boundary","wicket"]
//             //update the result based on the ball_result
//             if ball_result == "dot ball":
//                 print("dot ball")
//             else if ball_result == "single":
//                 current_score =+1   
//                 print("Single taken. Current_score:", current_score)
//             else if ball_result == "double":
//                 current_score += 2
//                 print("Double taken. Current_score:", current_score)
//             else if ball_result == "wicket":
//                 wickets += 1
//                 print("Wicket taken. wickets:", wickets)

//     return [current_score, wickets, balls, overs]
// }


fn main() {
    let mut raining = false;

    for overs in 1..=20 {
        println!("{} overs played", overs);

        if overs == 10 {
            raining = true;
            println!("It's raining, stopping play");
            break;
        }
    }

    if raining {
        println!("Resuming play after rain stops...");
        for overs in 11..=20 {
            println!("{} overs played", overs);
        }
    }
    println!("Game finished after 20 overs");
}
